### PR TITLE
docs: add regex/semver to versionFilter

### PR DESCRIPTION
**Pull request by mcwarman**
https://github.com/updatecli/updatecli/pull/3347

 - adds `regex/semver` to versionFilter page
 - adds dedicated version filter page
 - filter any pages that do not have title from search

## Test

This project uses Netlify to generate preview environment,
so feel free to look there directly to see how this pullrequest render

### CODE REVIEW

File: content/en/docs/core/versionFitler.adoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+title: "Version Filter"
```
Comment (mcwarman) [inline content/en/docs/core/versionFitler.adoc:2]: I always thought Version Filter should be its own page, it already already appears in the search but looks kind of broken. 

This tries to fix this.

File: content/en/docs/core/versionFilter.adoc [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+title: "Version Filter"
```
Comment (mcwarman) [inline content/en/docs/core/versionFilter.adoc:2]: I always thought Version Filter should be its own page, it already kind of is as it appears in the search but looks broken.

This tries to fix this.

Comment (mcwarman) [reply content/en/docs/core/versionFilter.adoc:2]: image

Comment (olblak) [reply content/en/docs/core/versionFilter.adoc:2]: I agree with you

**Review by olblak — APPROVED** [review]

**Pull request merged**